### DISCUSSION TITLE
Installing on PASE environment on IBMi 7.4 OS

### DISCUSSION THREAD

**Discussion by stopthatgoat** [ImageMagick/ImageMagick#3394] [Help]
I'm trying to get this installed on our PASE environment for the IBMi 7.4 OS.  It's not available via the standard IBM Open Source Management repository so checking whether there are any sort of installation assistance for the iseries PASE environment.  

Thanks for any help that can be provided.

**Comment by urban-warrior** [comment]
You can always build from source.  See https://imagemagick.org/script/install-source.php.

**Reply by stopthatgoat** [reply]
Well definitely getting further but now I'm getting stuck on the configure step.  

checking for stdlib.h... yes                                          
checking for string.h... yes                                          
checking for memory.h... yes                                          
checking for strings.h... yes                                         
checking for inttypes.h... yes                                        
checking for stdint.h... yes                                          
checking for unistd.h... yes                                          
checking minix/config.h usability... no                               
checking minix/config.h presence... no                                
checking for minix/config.h... no                                     
checking whether it is safe to define __EXTENSIONS__... yes           
checking for ar... ar                                                 
checking the archiver (ar) interface... unknown                       
configure: error: could not determine ar interface                    

Looks like the old ImageMagick website had some reports of this same problem for the IBMi but no solutions.

**Comment by BrianGodsend** [comment]
## DISCLAMER

**AT THE END OF A DAY, I STILL DID NOT GET THIS WORKING ... CLOSE (PERHAPS?), BUT, NOT WORKING.**

Following is what I did to get ImageMagick compiled (but not really working).  I simply ran out of time and patience and figure that if I cannot get it to be compiled and have the ability to decode a .jpg image after a full day of trying, then it is too complex for me to deploy in a production environment (with other developers less knowledgeable than me).  As such, I am abandoning my efforts, for now.  Hopefully, my notes on what I did will spur someone who actually knows what they are doing to take this to the finish line and post a comment here.

## Notes on my (failed) attempt

On an IBM i running v7.4 TR7, I just installed the Open Source Software (OSS) developer group using yum (`yum group install "Development Tools"`).  Also, based on a (this)[https://stackoverflow.com/a/66116056] post, I also installed the packages for:

* libjpeg8
* libpng-devel
* libpng16
* libtiff
* libtiff-devel
* libtiff5

Now that I had all the needed components, I then downloaded the current ImageMagick source release (v7.1.1-31) and unpacked it on the IFS.  When I ran `./configure --disable-shared`, I got the same error:

```
checking for ar... ar
checking the archiver (ar) interface... unknown
configure: error: could not determine ar interface
```
Looking in the log file, `config.log`, I found the following errors:

```
configure:6998: ar cru libconftest.a conftest.o >&5
ar: 0707-126 conftest.o is not valid with the current object file mode.
 Use the -X option to specify the desired object mode.
configure:7001: $? = 1
configure:7007: ar -NOLOGO -OUT:conftest.lib conftest.o >&5
ar: illegal option -- N
configure:7010: $? = 1
configure:7029: result: unknown
configure:7045: error: could not determine ar interface
```
It looks like the `ar` command on the IBM i requires the `-X` switch.  **I have no idea what this switch does, so I used the option _any_ ... this may be enough cause to stop trying what I did, again, I do not know what I am doing, other than trying to get ImageMagick to compile on my IBM i.**  To get the `configure` script to run, I edited the `configure` script file and changed:

`am_ar_try='$AR cru libconftest.a conftest.$ac_objext >&5'`
-to-
`am_ar_try='$AR -Xany cru libconftest.a conftest.$ac_objext >&5'`

I re-ran the `configure` script (with the `--disable-shared` option) and it completed normally.

Of course, when I ran the `make` command, it eventually ended with the same error thrown by the `ar` command.  So, I made changes to the following files:

* `./libtool`
   * I changed `lt_ar_flags=cr` to `lt_ar_flags='-Xany cr'`
* `./config/configure.xml`
   * I added  `<configure name="ARFLAGS" value="-Xand cr"/>` to the `<configuremap>` node

> **NOTE:**  
> Remember that if you run the `./configure` script it will (likely) replace these files.

Running `make` completed ... with a **lot** of warnings.  As they were only warnings (and too many for me do crawl through), I just moved on to the next step and ran `make install`.  This did get ImageMagick installed in `/usr/local/bin`.  However, when I tried a simple .jpg to .tiff conversion using `convert mypic.jpg mypic.tiff`, the command failed to decode the .jpg image.

So, at the end of a long day, I did get it to compile, but, it does not really work (at least for my needs) ... :frowning: ... `make uninstall` ... `signoff`,
